<a href="https://colab.research.google.com/github/JmB17/Perosnal-Project/blob/main/Potential_Datasets.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
import numpy as np
import pandas as pd

# Data Source

(https://www.kaggle.com/datasets/fedesoriano/cirrhosis-prediction-dataset)

Data Description:

Cirrhosis is a late stage of scarring (fibrosis) of the liver caused by many forms of liver diseases and conditions, such as hepatitis and chronic alcoholism. The following data contains the information collected from the Mayo Clinic trial in primary biliary cirrhosis (PBC) of the liver conducted between 1974 and 1984.

A total of 424 PBC patients, referred to Mayo Clinic during that ten-year interval, met eligibility criteria for the randomized placebo-controlled trial of the drug D-penicillamine. The first 312 cases in the dataset participated in the randomized trial and contain largely complete data. The additional 112 cases did not participate in the clinical trial but consented to have basic measurements recorded and to be followed for survival. Six of those cases were lost to follow-up shortly after diagnosis, so the data here are on an additional 106 cases as well as the 312 randomized participants.

In [18]:
df_cirr = pd.read_csv('/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week07/Data/cirrhosis.csv')
df_cirr.head(10)

ID  N_Days Status             Drug    Age Sex Ascites Hepatomegaly Spiders  \
0   1     400      D  D-penicillamine  21464   F       Y            Y       Y   
1   2    4500      C  D-penicillamine  20617   F       N            Y       Y   
2   3    1012      D  D-penicillamine  25594   M       N            N       N   
3   4    1925      D  D-penicillamine  19994   F       N            Y       Y   
4   5    1504     CL          Placebo  13918   F       N            Y       Y   
5   6    2503      D          Placebo  24201   F       N            Y       N   
6   7    1832      C          Placebo  20284   F       N            Y       N   
7   8    2466      D          Placebo  19379   F       N            N       N   
8   9    2400      D  D-penicillamine  15526   F       N            N       Y   
9  10      51      D          Placebo  25772   F       Y            N       Y   

  Edema  Bilirubin  Cholesterol  Albumin  Copper  Alk_Phos    SGOT  \
0     Y       14.5        261.0     2.60   156.0    1718.0  137.95   
1     N        1.1        302.0     4.14    54.0    7394.8  113.52   
2     S        1.4        176.0     3.48   210.0     516.0   96.10   
3     S        1.8        244.0     2.54    64.0    6121.8   60.63   
4     N        3.4        279.0     3.53   143.0     671.0  113.15   
5     N        0.8        248.0     3.98    50.0     944.0   93.00   
6     N        1.0        322.0     4.09    52.0     824.0   60.45   
7     N        0.3        280.0     4.00    52.0    4651.2   28.38   
8     N        3.2        562.0     3.08    79.0    2276.0  144.15   
9     Y       12.6        200.0     2.74   140.0     918.0  147.25   

   Tryglicerides  Platelets  Prothrombin  Stage  
0          172.0      190.0         12.2    4.0  
1           88.0      221.0         10.6    3.0  
2           55.0      151.0         12.0    4.0  
3           92.0      183.0         10.3    4.0  
4           72.0      136.0         10.9    3.0  
5           63.0        NaN         11.0    3.0  
6          213.0      204.0          9.7    3.0  
7          189.0      373.0         11.0    3.0  
8           88.0      251.0         11.0    2.0  
9          143.0      302.0         11.5    4.0

In [26]:
df_cirr.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 20 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   ID             418 non-null    int64  
 1   N_Days         418 non-null    int64  
 2   Status         418 non-null    object 
 3   Drug           312 non-null    object 
 4   Age            418 non-null    int64  
 5   Sex            418 non-null    object 
 6   Ascites        312 non-null    object 
 7   Hepatomegaly   312 non-null    object 
 8   Spiders        312 non-null    object 
 9   Edema          418 non-null    object 
 10  Bilirubin      418 non-null    float64
 11  Cholesterol    284 non-null    float64
 12  Albumin        418 non-null    float64
 13  Copper         310 non-null    float64
 14  Alk_Phos       312 non-null    float64
 15  SGOT           312 non-null    float64
 16  Tryglicerides  282 non-null    float64
 17  Platelets      407 non-null    float64
 18  Prothrombi

## Target

The target for this data set is our 'Stage' column, where it shows which stage of Cirrhosisvthey are at. Thius is on a scale from 1-4 with 4 being the worst stage of the disease.

## What does one row represent?

Each row represents each person who participated in the randomized trail and gives details on their age, sex, etc.

## Type of model

Since we are trying to predict what stage the person has of a disease its a classification model.

## Number of features

This data set here has a total of 20 features

## Number of rows

This dataset contains 418 rows

## What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

I like this data set alot and it has some good features and is clear on what is being predicted, I know I will face challanges with missing values and choosing which features to keep and use.

# Data Source

https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset?resource=download

Data Description:

According to the World Health Organization (WHO) stroke is the 2nd leading cause of death globally, responsible for approximately 11% of total deaths.

This dataset is used to predict whether a patient is likely to get stroke based on the input parameters like gender, age, various diseases, and smoking status. Each row in the data provides relavant information about the patient.

In [19]:
df_stroke = pd.read_csv('/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week07/Data/healthcare-dataset-stroke-data.csv')
df_stroke.head(10)

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   
5  56669    Male  81.0             0              0          Yes   
6  53882    Male  74.0             1              1          Yes   
7  10434  Female  69.0             0              0           No   
8  27419  Female  59.0             0              0          Yes   
9  60491  Female  78.0             0              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   
5        Private          Urban             186.21  29.0  formerly smoked   
6        Private          Rural              70.09  27.4     never smoked   
7        Private          Urban              94.39  22.8     never smoked   
8        Private          Rural              76.15   NaN          Unknown   
9        Private          Urban              58.57  24.2          Unknown   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1  
5       1  
6       1  
7       1  
8       1  
9       1

In [27]:
df_stroke.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


## Target

The target for this data set is the 'stroke' column which tells us if the patient had a stroke or not.

## What does one row represent?

Each row represents each patient and gives infomation like their age, if they ever had heart disease, if they caught a stroke before and etc.

## Type of model

Since we are trying to predict if a person will have a stroke, yes or no, its a classification model.

## Number of features

This data set here has a total of 12 features

## Number of rows

This dataset contains 5110 rows

## What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

I like this data set alot and it has some good features and is clear on what is being predicted, I'm excited to start the eda process and modeling for this data set.